src/object/srv_obj.c

```diff
@@ -2070,16 +2070,23 @@ ds_obj_ec_agg_handler(crt_rpc_t *rpc)
 	}
 	if (oea->ea_remove_nr) {
 		daos_epoch_range_t	epr;
+		uint64_t		stripe_end;
 		int			i;
 
+		stripe_end = (oea->ea_stripenum + 1) *
+			      oca->u.ec.e_len * oca->u.ec.e_k;
 		for (i = 0; i < oea->ea_remove_nr; i++) {
+			daos_recx_t *ea_recx;
+
+			ea_recx = &oea->ea_remove_recxs.ca_arrays[i];
+			if (DAOS_RECX_END(*ea_recx) > stripe_end)
```

Comment:
so does this extent never get deleted?   I didn't really grok the commit message.   It doesn't look like we delete the extent with current epoch but rather just skip it when the end spans more than one stripe

Reply:
I think that when processing next stripe, if the tail (exceed  part of last stripe) can be removed, then the original ext will be removed

Reply:
yes, the cross-stripe tail extent will be removed if the next stripe can fill the full stripe. otherwise it will be left there.